### PR TITLE
Move replace_java_nondet to java_bytecode 

### PR DESCRIPTION
**Pull request by peterschrammel**
and other small cleanups

### CODE REVIEW

**Review by kroening — APPROVED** [review]

**Comment by peterschrammel** [comment]
TG bump is passing.

File: src/ansi-c/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ various threading interfaces.
 
 \section preprocessing Preprocessing & Parsing
 
-In the \ref ansi-c and \ref java_bytecode directories
+In the \ref ansi-c directories
```
Comment (tautschnig) [inline src/ansi-c/README.md]: directory

File: src/ansi-c/README.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ digraph G {
 
 \section type-checking Type-checking
 
-In the \ref ansi-c and \ref java_bytecode directories.
+In the \ref ansi-c directories.
```
Comment (tautschnig) [inline src/ansi-c/README.md]: directory

File: src/java_bytecode/replace_java_nondet.cpp [hunk 3]
```diff
@@ -9,12 +9,13 @@ Author: Reuben Thomas, reuben.thomas@diffblue.com
...
-#include "goto-programs/goto_model.h"
-#include "goto-programs/remove_skip.h"
+#include "replace_java_nondet.h"
 
-#include "util/irep_ids.h"
+#include <goto-programs/goto_convert.h>
+#include <goto-programs/goto_model.h>
+#include <goto-programs/remove_skip.h>
+
+#include <util/irep_ids.h>
```
Comment (tautschnig) [inline src/java_bytecode/replace_java_nondet.cpp]: That’s hardly the right/necessary include here. std_code.h or maybe?

**Review by mgudemann — APPROVED** [review]
LGTM

**Pull request merged**